hivemind/p2p/p2p_daemon.py

```diff
@@ -147,15 +141,23 @@ async def replicate(cls, daemon_listen_port: int, host_port: int):
         # Use external already running p2pd
         self._child = None
         self._alive = True
-        self._assign_daemon_ports(host_port, daemon_listen_port)
+        self._assign_daemon_ports(p2p_port, daemon_listen_port)
         self._client_listen_port = find_open_port()
         self._client = p2pclient.Client(
             Multiaddr(f'/ip4/127.0.0.1/tcp/{self._daemon_listen_port}'),
             Multiaddr(f'/ip4/127.0.0.1/tcp/{self._client_listen_port}'))
         await self._wait_for_client()
```

Comment:
[not expected for this PR] this may be vulnerable to random failures where multiple peers request the same port which is open when they check for it, but then closen when they actually need it.